### PR TITLE
Add update component endpoint

### PR DESCRIPTION
**Pull request by yulintan**

### CODE REVIEW

File: component.go [hunk 1]
```diff
@@ -71,3 +71,34 @@ func (s *ComponentService) DeleteComponent(ctx context.Context, pageID string, c
 	_, err = s.client.do(ctx, req, nil)
 	return err
 }
+
+// UpdateComponentParams are the parameters that can be changed using the update component API endpoint
+type UpdateComponentParams struct {
+	Description        string    `json:"description,omitempty"`
+	Status             string    `json:"status,omitempty"`
+	Name               string    `json:"name,omitempty"`
+	OnlyShowIfDegraded string    `json:"only_show_if_degraded,omitempty"`
```
Comment (nagelflorian) [inline component.go]: Looking at the [API docs](https://developer.statuspage.io/#operation/patchPagesPageIdComponentsComponentId) this should be a boolean, no?

```suggestion
	OnlyShowIfDegraded bool      `json:"only_show_if_degraded,omitempty"`
```

File: component.go [hunk 2]
```diff
@@ -71,3 +71,34 @@ func (s *ComponentService) DeleteComponent(ctx context.Context, pageID string, c
...
+	GroupID            string    `json:"group_id,omitempty"`
+	Showcase           string    `json:"showcase,omitempty"`
```
Comment (nagelflorian) [inline component.go]: Looking at the [API docs](https://developer.statuspage.io/#operation/patchPagesPageIdComponentsComponentId) this should be a boolean, no?

```suggestion
	Showcase           bool      `json:"showcase,omitempty"`
```

**Review by nagelflorian — CHANGES_REQUESTED** [review]
Thanks for the contribution. :)

Changes look good to me. Just two small things.

File: component.go (hunk 1 above)
Comment (yulintan) [reply component.go]: Good catch! Refactored.

File: component.go (hunk 2 above)
Comment (yulintan) [reply component.go]: refactored

**Review by nagelflorian — APPROVED** [review]

**Comment by nagelflorian** [comment]
@yulintan Once again thanks for your contribution.

**Pull request merged**